sapi/fpm/tests/socket-uds-numeric-user-and-group-id.phpt

```diff
@@ -0,0 +1,48 @@
+--TEST--
+FPM: UNIX socket owner and group settings can be numeric
+--SKIPIF--
+<?php
+include "skipif.inc";
+FPM\Tester::skipIfNotRoot();
+?>
+--FILE--
+<?php
+
+require_once "tester.inc";
+
+$cfg = <<<EOT
+[global]
+error_log = {{FILE:LOG}}
+[unconfined]
+listen = {{ADDR:UDS}}
+listen.owner = 1234
+listen.group = 1234
+user = 1234
+ping.path = /ping
+ping.response = pong
+pm = dynamic
+pm.max_children = 5
+pm.start_servers = 2
+pm.min_spare_servers = 1
+pm.max_spare_servers = 3
+EOT;
+
+$tester = new FPM\Tester($cfg);
+$tester->start();
```

Comment:
I just tested this to run under root and it actually fails to start for me. Might be my local issue but was wondering if the test actually works for you when you run it under root?

Reply:
Btw for me this line fails if run under root even with setting `TEST_FPM_RUN_AS_ROOT=1`:
```
$this->masterProcess = proc_open($cmd, $desc, $pipes);
```
Interestingly I tested starting that with passthru and it actually starts FPM so something weird with `proc_open`. Might need to debug it.

Reply:
It does work for me:

```
$ sudo -i
# cd /home/andre/git/forks/php-src
# TEST_FPM_RUN_AS_ROOT=1 ./sapi/cli/php run-tests.php sapi/fpm/tests/socket-uds-numeric-user-and-group-id.phpt

=====================================================================
PHP         : /home/andre/git/forks/php-src/sapi/cli/php 
PHP_SAPI    : cli
PHP_VERSION : 8.0.0-dev
ZEND_VERSION: 4.0.0-dev
PHP_OS      : Linux - Linux andre 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64
INI actual  : /home/andre/git/forks/php-src
More .INIs  :   
---------------------------------------------------------------------
PHP         : /home/andre/git/forks/php-src/sapi/phpdbg/phpdbg 
PHP_SAPI    : phpdbg
PHP_VERSION : 8.0.0-dev
ZEND_VERSION: 4.0.0-dev
PHP_OS      : Linux - Linux andre 4.15.0-88-generic #88-Ubuntu SMP Tue Feb 11 20:11:34 UTC 2020 x86_64
INI actual  : /home/andre/git/forks/php-src
More .INIs  : 
---------------------------------------------------------------------
CWD         : /home/andre/git/forks/php-src
Extra dirs  : 
VALGRIND    : Not used
=====================================================================
Running selected tests.
PASS FPM: UNIX socket owner and group settings can be numeric [sapi/fpm/tests/socket-uds-numeric-user-and-group-id.phpt] 
=====================================================================
Number of tests :    1                 1
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Tests passed    :    1 (100.0%) (100.0%)
---------------------------------------------------------------------
Time taken      :    0 seconds
=====================================================================
```

The test calls the `stat` command to check the owner and group of the created sockets. Could it be a `PATH` issue, or maybe your environment doesn't have this command?

Reply:
Probably something local then. Don't worry about it then!